catalog/files/impl/src/main/java/org/projectnessie/catalog/files/adls/AdlsClientSupplier.java

```diff
@@ -89,10 +90,7 @@ DataLakeFileSystemClient fileSystemClient(AdlsLocation location) {
       String accountKey = fileSystemOptions.account().get().secret();
       clientBuilder.credential(new StorageSharedKeyCredential(accountName, accountKey));
     } else {
-      throw new IllegalStateException(
-          "Neither an SAS token nor account name and key are available for ADLS file system '"
-              + location.path()
-              + "'");
+      clientBuilder.credential(new DefaultAzureCredentialBuilder().build());
```

Comment:
@cgpoh thank you for your PR!

Very recently, we adopted a new configuration strategy for S3 and GCS: instead of trying options one by one like here, we introduced a config option like this one for GCS:

https://github.com/projectnessie/nessie/blob/e58d579ef68ea907f3d3fa2e00ff7472ca3150c0/catalog/files/impl/src/main/java/org/projectnessie/catalog/files/gcs/GcsBucketOptions.java#L54

This option is then processed like this:

https://github.com/projectnessie/nessie/blob/b6187e2cbe7a92bdfc85a8ddfb65a26b30ab3a00/catalog/files/impl/src/main/java/org/projectnessie/catalog/files/gcs/GcsClients.java#L123-L171

I think it could be cool to do the same for ADLS, WDYT?

Reply:
@adutra , thanks for the review. I see whether I can do the same for ADLS.